src/help.js

```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
   }
 }
 var stringFormatting = function stringFormatting (str) {
-  const rights = /\(privileged: admins only\)/
-  str = '*' + str
-  str = str.replace(/ - /i, ' *- ')
-  return str.replace(rights, '`privileged: admins only`')
+  if (!str.match(/^(begin|end)/i)) {
+    str = '*' + str
+    str = str.replace(/ - /i, ' *- ')
+  }
+  return str
+}
+
+/**
+ * parses an array with commands and collects an object with keys called the group name.
+ *
+ * @param {Robot} robot - Hubot instance.
+ * @param {array} commands - Array of commands.
+ * @param {string} markerName - The name of the marker that will initialize a group of commands.
+ *
+ * @returns {object}
+ */
+var groupingByCommands = function groupingByCommands (robot, commands, markerName) {
+  let err
+  let inBegin = false
+  let inGroup = false
+  let groupCommand = {}
+  let marker
+  let markerGroup
+  let markerReg = new RegExp(`(begin|end)\\s*(${markerName})\\s*(.*)`, 'i')
+
+  let pushInGroup = (command, markerGroup) => {
+    if (!groupCommand[markerGroup]) {
+      groupCommand[markerGroup] = []
+    }
+    groupCommand[markerGroup].push(command)
+  }
+
+  commands.map((command, index) => {
+    marker = command.match(markerReg)
+    err = `${markerGroup ? `in the script "${markerGroup}" is` : `in some script the`} closing marker was not added`
+    if (marker) {
+      if ((inGroup && (marker[1] + marker[2]) === 'begingroup')) {
+        throw (routines.rave(robot, err))
+      }
+      if (!inBegin && marker[0] === `end ${markerName}`) {
+        markerGroup = !inBegin ? false : markerGroup
+        err = `${markerGroup ? `in the script "${markerGroup}"` : `in some script the`} opening marker was not added`
+        throw (routines.rave(robot, err))
+      }
+
+      inBegin = marker[1] === 'begin'
+      inGroup = inBegin && marker[2] === markerName
+      markerGroup = (inGroup && marker[0]) ? marker[3] : markerGroup
+    } else {
+      if (inGroup) {
+        if (commands.length === index + 1) {
+          throw (routines.rave(robot, err))
+        }
+        pushInGroup(command, markerGroup)
+      } else {
+        pushInGroup(command, 'Other commands')
+      }
+    }
+  })
+  return groupCommand
+}
+
+/**
+ * Constructs a rich messages.
```

Comment:
`* Constructs the array of rich messages.`
And you need to describe the `groupCommand` argument.